### PR TITLE
[docs] Update Adding React Components

### PR DESCRIPTION
**Pull request by AishaBlake**
## Description

Having gone through a workflow evaluation for working with style libraries and components, we identified some useful updates for the page on Adding React Components. These are listed in issue #21278. 

## Related Issues

Addresses #21278

### CODE REVIEW

File: docs/docs/adding-react-components.md [hunk 1]
```diff
@@ -2,13 +2,13 @@
 title: Adding React Components
 ---
 
-This guide covers how to add React components to your Gatsby site.
+This guide covers how to add React components, including those from third-party component libraries, to your Gatsby site.
```
Comment (laurieontech) [inline docs/docs/adding-react-components.md]:
```suggestion
This guide covers how to add React components to your Gatsby site, including those from third-party component libraries.
```

File: docs/docs/adding-react-components.md [hunk 2]
```diff
@@ -2,13 +2,13 @@
...
+React components are prebuilt elements or groups of elements that can be used to split your user interface (UI) into independent, reusable pieces. There are multiple types of components you can write but this guide covers functional components. For more in-depth information on writing React components, including classes, check out the [React documentation](https://reactjs.org/docs/components-and-props.html).
 
-Components also offer the ability to be customized using inputs, better known as "props" (properties). Props can be of any JavaScript type, such as Boolean, String, Object, Array or almost anything you can think of.
+Components can also be customized using inputs, better known as "props" (properties). Props can be of any JavaScript data type, such as Boolean, String, Object, or Array.
```
Comment (laurieontech) [inline docs/docs/adding-react-components.md]:
```suggestion
Components can also be customized using inputs, better known as "props" (properties). Props can be of any JavaScript data type, such as Boolean, String, or Object.
```

File: docs/docs/adding-react-components.md [hunk 3]
```diff
@@ -29,19 +29,27 @@ export default () => (
 
 ### Importing third-party components
 
-Just like React, Gatsby also supports third-party components and libraries. You can install a third-party component or library via your package manager. We tend to favour and use npm and we will reflect this in our examples.
+Just like React, Gatsby also supports third-party components and libraries. You can install a third-party component or library via your package manager. We tend to favor and use npm and we will reflect this in our examples.
 
 Here's an example of adding a third-party component to your site.
 
-First, you have to install the component or library's package via a package manager. It's recommended not to mix package managers, so if you use npm, don't use another and vice versa.
+First, you have to install the component or library's package via a package manager. It's recommended not to mix package managers, so if you use npm, don't use another and vice versa. If there's a relevant Gatsby plugin, you should install and use that plugin.
```
Comment (laurieontech) [inline docs/docs/adding-react-components.md]:
```suggestion
First, you have to install the component or library's package via a package manager. It's recommended not to mix package managers, so if you use npm, don't use another and vice versa. If there's a relevant Gatsby plugin, you should install and use that plugin.

In this example, you'll install both a plugin and the library it depends on:
```

File: docs/docs/adding-react-components.md [hunk 4]
```diff
@@ -29,19 +29,27 @@ export default () => (
...
-```jsx:title=my-page.jsx
+```js:title=gatsby-config.js
+module.exports = {
+  plugins: [`gatsby-plugin-material-ui`],
+};
+```
+
+Import and use the library in your page's source:
+
+```jsx:title=index.jsx
```
Comment (laurieontech) [inline docs/docs/adding-react-components.md]:
```suggestion
```jsx:title=index.jsx
```

In order for the button to appear without being pressed we should make the button ```
    <Button variant="contained">Fancy button!</Button>

```

**Comment by AishaBlake** [comment]
@laurieontech I think this is ready for another look!

File: docs/docs/adding-react-components.md [hunk 5]
```diff
@@ -29,19 +29,29 @@ export default () => (
...
+module.exports = {
+  plugins: [`gatsby-plugin-material-ui`],
+};
 ```
 
-After you've installed a package, import and use it in your page's source:
+Import and use the library in your page's source:
 
-```jsx:title=my-page.jsx
+```jsx:title=index.jsx
```
Comment (laurieontech) [inline docs/docs/adding-react-components.md]:
```suggestion
```jsx:title=index.js
```
Realizing is file name is incorrect.

**Comment by laurieontech** [comment]
Looks great. Thanks Aisha!

**Pull request merged**